cl/citations/tests.py

```diff
@@ -2161,3 +2195,54 @@ async def test_can_extract_all_citations_within_text(self) -> None:
 
         clusters = second_citation["clusters"]
         self.assertEqual(len(clusters), 0)
+
+    @override_settings(MAX_CITATIONS_PER_REQUEST=10)
+    async def test_can_limit_max_citations_per_request(
+        self, cache_key_mock
+    ) -> None:
+        ten_citations = "56 F.2d 9, " * 10
+        text_citation = f"{ten_citations} 139 U.S. 601, 155 U.S. 597"
+        r = await self.async_client.post(
+            reverse("citation-lookup-list", kwargs={"version": "v3"}),
+            {"text": text_citation},
+        )
+
+        self.assertEqual(r.status_code, HTTPStatus.OK)
+        data = json.loads(r.content)
+        self.assertEqual(len(data), 11)
+
+        last_citation = data[-1]
+        self.assertEqual(last_citation["citation"], "139 U.S. 601")
+        self.assertEqual(last_citation["status"], HTTPStatus.TOO_MANY_REQUESTS)
+        self.assertEqual(last_citation["error_message"], "Too many requests.")
+
+    @patch(
+        "cl.api.utils.CitationCountRateThrottle.get_citations_rate",
+        return_value=(20, 60),
+    )
+    async def test_can_rate_limit_user_based_on_citation_count(
+        self, get_rate_mock, throttle_logic_mock
+    ) -> None:
+        throttle_logic_mock.return_value = "citation_throttle_test"
+        sixty_citations = "56 F.2d 9, " * 60
+        r = await self.async_client.post(
+            reverse("citation-lookup-list", kwargs={"version": "v3"}),
+            {"text": sixty_citations},
+        )
+
+        self.assertEqual(r.status_code, HTTPStatus.OK)
+        data = json.loads(r.content)
+        self.assertEqual(len(data), 60)
+
+        # This test only allows 20 citations per minute, but the last request
+        # had 60. This request should be rate limited.
+        ten_citations = "56 F.2d 9, " * 10
+        r = await self.async_client.post(
+            reverse("citation-lookup-list", kwargs={"version": "v3"}),
+            {"text": ten_citations},
+        )
+
+        self.assertEqual(r.status_code, HTTPStatus.TOO_MANY_REQUESTS)
+        data = json.loads(r.content)
+        self.assertEqual(data["error_message"], "Request was throttled.")
+        self.assertIn("wait_until", data)
```

Comment:
I think it'd be helpful to actually have an assertion for the correct value of `wait_until`. I think if you use [time machine](https://adamj.eu/tech/2020/06/03/introducing-time-machine/), you can accomplish things like this? I think we already have it as a dependency?

Reply:
> I think we already have it as a dependency?

Yes, It's already a dependency. I'll update the test.